### PR TITLE
golang filter: parse plugin config in the golang side while initializing golang filter

### PR DESCRIPTION
**Pull request by doujiang24**
it will reject the config when failed to parse plugin config in the golang side.

fix https://github.com/envoyproxy/envoy/issues/25369

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: contrib/golang/filters/http/source/golang_filter.cc [hunk 1]
```diff
@@ -1176,40 +1172,70 @@ FilterConfigPerRoute::FilterConfigPerRoute(
...
   return parent_id;
 }
 
-uint64_t RoutePluginConfig::getMergedConfigId(uint64_t parent_id, std::string so_id) {
+RoutePluginConfig::RoutePluginConfig(
+    const std::string plugin_name,
+    const envoy::extensions::filters::http::golang::v3alpha::RouterPlugin& config)
+    : plugin_name_(plugin_name), plugin_config_(config.config()) {
+
+  ENVOY_LOG(debug, "initilizing golang filter route plugin config, plugin_name: {}, type_url: {}",
```
Comment (wangfakang) [inline contrib/golang/filters/http/source/golang_filter.cc:1191]: typo: initilizing -> initializing

**Review by wangfakang — COMMENTED** [review]
some minor issue, others LGTM. Thanks.

**Comment by doujiang24** [comment]
oh, seems the [coverage failure ](https://dev.azure.com/cncf/envoy/_build/results?buildId=139516&view=logs&j=bbe4b42d-86e6-5e9c-8a0b-fea01d818a24&t=299ff00a-2cef-5552-88aa-ea9cf69316f0&l=5059)is introduce in https://github.com/envoyproxy/envoy/pull/27822
I can see the master branch also have [such failure](https://dev.azure.com/cncf/envoy/_build/results?buildId=139501&view=logs&j=bbe4b42d-86e6-5e9c-8a0b-fea01d818a24&t=299ff00a-2cef-5552-88aa-ea9cf69316f0&l=5053), but https://github.com/envoyproxy/envoy/pull/27822 is passed, so I'm a bit confused.

@alyssawilk could you please take a look? thanks!

**Comment by alyssawilk** [comment]
thanks for the report.  tracking on https://github.com/envoyproxy/envoy/issues/27844 - I'll try to get a fix or revert landed today

**Review by mattklein123 — APPROVED** [review]

**Comment by mattklein123** [comment]
Will wait for CI to be sorted.

/wait

**Pull request merged**